### PR TITLE
Add support for Avg aggregation

### PR DESCRIPTION
**Pull request by Mitep**

### CODE REVIEW

File: modules/library/src/main/scala/zio/elasticsearch/aggregation/Aggregations.scala [hunk 1]
```diff
@@ -31,10 +31,30 @@ sealed trait ElasticAggregation { self =>
 
 sealed trait SingleElasticAggregation extends ElasticAggregation
 
+sealed trait AvgAggregation extends SingleElasticAggregation with HasMissing[AvgAggregation] with WithAgg
+
+private[elasticsearch] final case class Avg(name: String, field: String, missing: Option[Double])
+    extends AvgAggregation {
+  self =>
```
Comment (dbulaja98) [inline modules/library/src/main/scala/zio/elasticsearch/aggregation/Aggregations.scala:38]: Can you move `self` up, to 37. line?

File: modules/library/src/main/scala/zio/elasticsearch/aggregation/Aggregations.scala [hunk 2]
```diff
@@ -31,10 +31,30 @@ sealed trait ElasticAggregation { self =>
...
+
+  def withAgg(agg: SingleElasticAggregation): MultipleAggregations =
+    multipleAggregations.aggregations(self, agg)
+
+  def missing(value: Double): AvgAggregation =
+    self.copy(missing = Some(value))
```
Comment (dbulaja98) [inline modules/library/src/main/scala/zio/elasticsearch/aggregation/Aggregations.scala:44]: Can you sort this?

File: modules/library/src/main/scala/zio/elasticsearch/aggregation/Aggregations.scala [hunk 3]
```diff
@@ -31,10 +31,30 @@ sealed trait ElasticAggregation { self =>
...
+    Obj(name -> Obj("avg" -> (Obj("field" -> field.toJson) merge missingJson)))
+  }
+}
+
 sealed trait BucketSelectorAggregation extends SingleElasticAggregation with WithAgg
 
 private[elasticsearch] final case class BucketSelector(name: String, script: Script, bucketsPath: Map[String, String])
-    extends BucketSelectorAggregation { self =>
+    extends BucketSelectorAggregation {
+  self =>
```
Comment (dbulaja98) [inline modules/library/src/main/scala/zio/elasticsearch/aggregation/Aggregations.scala:57]: Can you move self up?

File: modules/library/src/main/scala/zio/elasticsearch/result/AggregationResult.scala [hunk 4]
```diff
@@ -21,6 +21,7 @@ import zio.Chunk
 import scala.util.{Failure, Success, Try}
 
 sealed trait AggregationResult
+final case class AvgAggregationResult private[elasticsearch] (value: Double) extends AggregationResult
```
Comment (dbulaja98) [inline modules/library/src/main/scala/zio/elasticsearch/result/AggregationResult.scala:24]: Can you leave one empty line between?

**Review by dbulaja98 — CHANGES_REQUESTED** [review]

File: modules/library/src/main/scala/zio/elasticsearch/aggregation/Aggregations.scala (hunk 2 above)
Comment (markaya) [inline modules/library/src/main/scala/zio/elasticsearch/aggregation/Aggregations.scala]: order methods by alaphabet

**Review by markaya — CHANGES_REQUESTED** [review]

**Review by dbulaja98 — APPROVED** [review]

**Review by markaya — APPROVED** [review]

**Pull request merged**